lib/internal/webidl.js

```diff
@@ -209,10 +220,77 @@ function createEnumConverter(name, values) {
   };
 }
 
+// https://tc39.es/ecma262/#sec-ecmascript-data-types-and-values
+function type(V) {
+  if (V === null)
+    return 'Null';
+
+  switch (typeof V) {
+    case 'undefined':
+      return 'Undefined';
+    case 'boolean':
+      return 'Boolean';
+    case 'number':
+      return 'Number';
+    case 'string':
+      return 'String';
+    case 'symbol':
+      return 'Symbol';
+    case 'bigint':
+      return 'BigInt';
+    case 'object': // Fall through
+    case 'function': // Fall through
+    default:
+      // Per ES spec, typeof returns an implemention-defined value that is not
+      // any of the existing ones for uncallable non-standard exotic objects.
+      // Yet Type() which the Web IDL spec depends on returns Object for such
+      // cases. So treat the default case as an object.
+      return 'Object';
+  }
+}
+
+// https://webidl.spec.whatwg.org/#es-sequence
+// TODO: make this completely spec compliant
```

Comment:
What does this comment mean?

Reply:
I thought it was not compliant with the spec with my first look, but I was wrong. It's compliant after double checking the spec. I'll remove this. 👍